### PR TITLE
[Hub Generated] Review request for Microsoft.ConfidentialLedger to add version preview/2023-01-26-preview

### PR DESCRIPTION
**Pull request by musabbir**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/musabbir-confidentialledger-Microsoft.ConfidentialLedger-2023-01-26-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.

  February, 2023

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  
  February 2023

  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sjanamma** [comment]
@musabbir - Please see if you can fix all the Linter warnings.
https://github.com/Azure/azure-rest-api-specs/pull/22325/checks?check_run_id=10916052276
https://github.com/Azure/azure-rest-api-specs/pull/22325/checks?check_run_id=10916052374

There is also a BreakingChangeReviewRequired label on the PR. You have an option to fix your contracts to be backward compatible or reach out to azbreakchangereview@microsoft.com get an excepton

**Comment by musabbir** [comment]
> @musabbir - Please see if you can fix all the Linter warnings. https://github.com/Azure/azure-rest-api-specs/pull/22325/checks?check_run_id=10916052276 https://github.com/Azure/azure-rest-api-specs/pull/22325/checks?check_run_id=10916052374
> 
> There is also a BreakingChangeReviewRequired label on the PR. You have an option to fix your contracts to be backward compatible or reach out to [azbreakchangereview@microsoft.com](mailto:azbreakchangereview@microsoft.com) get an excepton

Got the BreakingChange approval.
I fixed couple of the linter warnings. The remaining ones are part of a private preview feature and is out of scope for the current change. Those will be taken into account when that feature moves to public preview later this year.

**Review by leniatgh — APPROVED** [review]

**Comment by leni-msft** [comment]
@musabbir It seems the new api-version is not deployed yet. Since your RP is based on RPaaS, please go ahead with the deployment after the PR is merged.

**Comment by leni-msft** [comment]
@tadelesh @msyyc please help review the SDK breaking changes

**Comment by tadelesh** [comment]
> @tadelesh @msyyc please help review the SDK breaking changes

@musabbir Please consolidate the `ErrorResponse` common type version.

**Comment by musabbir** [comment]
> @musabbir It seems the new api-version is not deployed yet. Since your RP is based on RPaaS, please go ahead with the deployment after the PR is merged.

Yes that is the plan. I am waiting for this PR to be merged.

**Comment by musabbir** [comment]
> > @tadelesh @msyyc please help review the SDK breaking changes
> 
> @musabbir Please consolidate the `ErrorResponse` common type version.

I updated confidential ledger to use v3 same as managed ccf.

**Comment by tadelesh** [comment]
@musabbir There are still duplicate schema issue of `Resource`. Could you please refer this [doc](https://github.com/Azure/azure-rest-api-specs/wiki/Common-CI-Error#duplicated-schema) to fix it?

**Comment by musabbir** [comment]
> @musabbir There are still duplicate schema issue of `Resource`. Could you please refer this [doc](https://github.com/Azure/azure-rest-api-specs/wiki/Common-CI-Error#duplicated-schema) to fix it?

Thank you for the link. I fixed that issue and I can see majority of the sdks building now.
Not quite sure why `js` and `powershell` is broken however. Can you point me in the right direction or can these be ignored?

**Comment by tadelesh** [comment]
@MaryGao @qiaozha Could you help with JS SDK generation failure?

File: specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json
Comment (MaryGao) [inline specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json:422]: Here the property `location` is optional.

File: specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json
Comment (MaryGao) [inline specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json:663]: To fix this you could remove the ResourceLocation or set its location as required.

The property `location` is defined in both TrackedResource and ResourceLocation. But the definition has conflict with one other cause one is required but the other is optional.

https://github.com/Azure/azure-rest-api-specs/blob/d883967b5cc43f9a7fb9c53d0fb3c36bcc373da9/specification/common-types/resource-management/v3/types.json#L73-L84


https://github.com/Azure/azure-rest-api-specs/blob/d883967b5cc43f9a7fb9c53d0fb3c36bcc373da9/specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json#L421-L429

Comment (musabbir) [reply specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json]: Thanks for the pointer. I removed `ResourceLocation` and JS is now building.

**Comment by musabbir** [comment]
@leni-msft
Is this okay to merge now?

**Pull request merged**
